vm/src/frame.rs

```diff
@@ -1659,6 +1699,14 @@ impl ExecutingFrame<'_> {
             vm.ctx.new_dict().into()
         };
 
+        let type_params: PyTupleRef = if flags.contains(bytecode::MakeFunctionFlags::TYPE_PARAMS) {
+            self.pop_value()
+                .downcast()
+                .map_err(|_| vm.new_type_error("Type params must be a tuple.".to_owned()))?
+        } else {
+            vm.ctx.new_tuple(vec![])
```

Comment:
```suggestion
            vm.ctx.empty_tuple.clone()
```
empty_tuple shares a singleton object